packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -451,16 +451,21 @@ export class TokenDetectionController extends StaticIntervalPollingController<
     }
     const isTokenDetectionInactiveInMainnet =
       !this.#isDetectionEnabledFromPreferences && chainId === ChainId.mainnet;
-    const { tokenList } = this.messagingSystem.call(
+    const { tokensChainsCache } = this.messagingSystem.call(
       'TokenListController:getState',
     );
+    const tokenList = tokensChainsCache[chainId]?.data || {};
```

Comment:
Nit: What do you think about using the `??` operator here? 

```suggestion
    const tokenList = tokensChainsCache[chainId]?.data ?? {};
```

Unless there's a falsy value of the left-hand side that we're trying to account for, `??` would be more type safe and readable (less possibilities to consider for code reviewer).

Providing a default value for property access that uses optional chaining also happens to be a use case for `??` [specified in MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing#relationship_with_the_optional_chaining_operator).